lib/Resizable/index.tsx

```diff
@@ -0,0 +1,123 @@
+import type { PropsWithChildren } from "react";
+import React, { useCallback, useEffect, useRef } from "react";
+import { keepValueWithinRange, toPixels } from "../helpers";
+
+export interface ResizableProps {
+  defaultWidth?: number | string;
+  minWidth?: number | string;
+  maxWidth?: number | string;
+  useGutterOffset?: boolean;
+  containerWidth?: number | string;
+}
+
+/**
+ * A wrapper for making a given child element resizable
+ * @param children The element to be resized
```

Comment:
What's the reasoning for having types and comment blocks 🤔 

They are already out of sync and like in this case shouldn't `children` be added to the interface but it isn't it's only described in the comment block? 🤔

Reply:
For me comment blocks are currently used to explain a complex or smelly piece of implementation, I don't see how the comment blocks add anything other than a maintenance overhead for future refactors. Some ideas to remove the need for the comment blocks...

-  `maxWidth` to `maxResizableWidth`
-  `minWidth` to `minResizedWidth`
- `defaultWidth` to `initialWidth`
- add type for `children` as I am assuming it can't be any react node (e.g. a fragment)

Reply:
Thanks Kyle. I think at the time I was having a bit of trouble trying to remember what props I needed and why I needed them as I was working around new problems and iterating. I'll take them out and refactor the props & interface as advised now the solution is "complete".

Reply:
Just had a look back at the code to refresh my memory and we are actually defining the `children` prop via PropsWithChildren: `}: PropsWithChildren<ResizableProps>) {`